src/clj.lfe

```diff
@@ -527,6 +527,20 @@
   sequence."
   (lists:seq start end step))
 
+(defmacro conj
+  "conj[oins] a value onto an existing collection, either a list, a tuple,
+ or a map. For lists this means prepending, for tuples appending,
+ and for maps merging."
+  (`[,coll . ,xs]
+   `(cond ((is_list ,coll) (cons ,@xs ,coll))
+          ((is_tuple ,coll)
+           (lists:foldl (lambda (x acc) (erlang:append_element acc x))
```

Comment:
I seem to recall wasting a lot of time converting types back and forth initially, so so far this has been the best solution I could come up with.